### PR TITLE
add backport of master's m_conn_join

### PR DESCRIPTION
**Pull request by Sheogorath-SI**
After it was requested multiple times, I submit it to extras now so everyone can use it.

### CODE REVIEW

File: 2.0/m_conn_join_backport.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ 
+		if (user->chans.empty())
+			JoinChannels(user, channels);
+	}
+};
+
+class ModuleConnJoin : public Module
+{
+	public:
+        ModuleConnJoin()
```
Comment (SaberUK) [inline 2.0/m_conn_join_backport.cpp]: An empty default constructor is unnecessary.

File: 2.0/m_conn_join_backport.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		{
+		}
+
+		void init()
+		{
+            OnRehash(NULL);
```
Comment (SaberUK) [inline 2.0/m_conn_join_backport.cpp]: Indentation is broken.

File: 2.0/m_conn_join_backport.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ 
+			if (!chandelay)
+				JoinChannels(localuser, chanlist);
+			else
+				new JoinTimer(localuser, chanlist, chandelay);
+		}
+
+		void OnRehash(User* user)
+		{
+			if (ServerInstance->Modules->Find("m_conn_join.so") != NULL)
```
Comment (SaberUK) [inline 2.0/m_conn_join_backport.cpp]: OnModuleLoad might be a better place to do this.

**Comment by Sheogorath-SI** [comment]
Changes are done. Any further improvements?

File: 2.0/m_conn_join_backport.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+			JoinChannels(user, channels);
+	}
+};
+
+class ModuleConnJoin : public Module
+{
+	public:
+		void init()
+		{
+			OnRehash(NULL);
```
Comment (Adam-) [inline 2.0/m_conn_join_backport.cpp]: Also check if m_conn_join is loaded here

**Comment by Adam-** [comment]
Maybe also rename the module to reflect what it does.

**Comment by Sheogorath-SI** [comment]
Any further improvements?

**Pull request merged**